test/object-comprehensions.civet

```diff
@@ -0,0 +1,118 @@
+{ testCase } from ./helper.civet
+
+describe "object comprehensions", ->
+  testCase """
+    basic
+    ---
+    {
+      for x of [1, 2, 3]
+        [x]:  2 * x
+        [x * 2]: 4 * x
+    }
+    ---
+    ({
+      ...(()=>{const results={};for (const x of [1, 2, 3]) {
+        Object.assign(results,({[x]:  2 * x,
+        [x * 2]: 4 * x}))
+      }return results})()
+    })
+  """
```

Comment:
An interesting issue is that [TypeScript completely gets the types wrong here](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBCMF4bIBQG8CwAoZuYDpCUUBKBAPjVElgCcBTCAVwBsoIE0BfAbgDMQtGCmrQYADzh8YAbQCMAGhgAmJQGYAuiRiYceXAHkARgCt6wKPgCGECAEsA5mBQNmbCAvQzxGgFzJlGAAqCQVsfVxvYJU-GAAWaPEuEhJw-S4GKCZaMBhXVnZk0jTkZOw0kGwAeiqAPQB+IA); I guess it just ignores `Object.assign`. At least it doesn't error, but it's sad that it doesn't figure out that it's `Record<string, number>`.

I don't know a way to fix this. (I tried `let results;results={}` but it behaved the same.) I guess we can always explicit typecast, e.g. like this:
```js
    {
      for x of [1, 2, 3]
        [x]:  2 * x
        [x * 2]: 4 * x
    } as Record<string, number>
```